pydantic_ai_slim/pydantic_ai/models/google.py

```diff
@@ -565,17 +566,46 @@ async def _map_user_prompt(self, part: UserPromptPart) -> list[PartDict]:
                 if isinstance(item, str):
                     content.append({'text': item})
                 elif isinstance(item, BinaryContent):
-                    inline_data_dict: BlobDict = {'data': item.data, 'mime_type': item.media_type}
-                    part_dict: PartDict = {'inline_data': inline_data_dict}
-                    if item.vendor_metadata:
-                        part_dict['video_metadata'] = cast(VideoMetadataDict, item.vendor_metadata)
-                    content.append(part_dict)
+                    if BinaryContent.is_text_like_media_type(item.media_type):
+                        content.append(
+                            BinaryContent.inline_text_file_part(
```

Comment:
The error you asked about in the other thread is because `content` is a `list` of `PartDict`s, but `inline_text_file_part` returns a `str`. Like on line 567, you need to wrap it in `{'text': ...}`.

Note that if you were running a type checker like pyright in your IDE or manually, it would have told you this.